src/cljs/athens/electron.cljs

```diff
@@ -323,6 +324,55 @@
   ;; open or create a new starter db
 
   ;; Watch filesystem, e.g. in case db is updated via Dropbox sync
+  
+  (def schemaTest
+  {:schema/version {}
+   :block/uid      {:db/unique :db.unique/identity}
+   :node/title     {:db/unique :db.unique/identity}
+   :attrs/lookup   {:db/cardinality :db.cardinality/many}
+   :block/children {:db/cardinality :db.cardinality/many
+                    :db/valueType :db.type/ref}
+   :block/refs     {:db/cardinality :db.cardinality/many
+                    :db/valueType :db.type/ref}}
+  )
+  
+  ;; Create an empty Db for test 
+  (def test-db
+    (-> (d/empty-db schemaTest))
+  )
+  
+  ;; Update index.transit
+  (defn writeDbIndex [file filepath]
+    (let [w  (.. fs (writeFile filepath file (fn [e](dispatch [:boot/desktop]))))])
+  )
+  
+  
+  (defn open-dialog-index
+    "Allow user to open a Backup file."
+    [filepath]
+    (js/alert "Your Data file is corrupted or incorrect, Please select a Backup file")
+    (let [res  (.showOpenDialogSync dialog (clj->js {:properties ["openFile"]
+                                                     :filters    [{:name "Transit" :extensions ["bkp"]}]}))
+          open-file (first res)]
+      (when (and open-file (.existsSync fs open-file))
+        (let [read-db (.readFileSync fs open-file)
+              db      (try  (dt/read-transit-str read-db)(catch  :default e (open-dialog-index filepath)))              
+              ]  
+           (if (when (= (:schema db)  (:schema test-db)) true) 
+              ((writeDbIndex read-db filepath))((open-dialog-index filepath))))))
```

Comment:
does this actually work? this looks to me like too many braces. `(write-db-index read-db filepath)` is a function call to `write-db-index`. `((write-db-index read-db filepath))` is a function call of the *result* of a function call to `write-db-index`.

Reply:
it works this way if you remove the braces , then it will loop many times before validating the copy file even if the file if already available and it allows you to call the :reboot callback function